app/src/ui/history/commit-list.tsx

```diff
@@ -8,26 +8,52 @@ import { IGitHubUser } from '../../lib/databases'
 const RowHeight = 50
 
 interface ICommitListProps {
-  readonly onCommitChanged: (commit: Commit) => void
-  readonly onScroll: (start: number, end: number) => void
-  readonly onRevertCommit: (commit: Commit) => void
-  readonly onViewCommitOnGitHub: (sha: string) => void
-  readonly repository: Repository
-  readonly history: ReadonlyArray<string>
-  readonly commits: Map<string, Commit>
+  /** The GitHub repository associated with this commit (if found) */
+  readonly gitHubRepository: GitHubRepository | null
+
+  /** The list of commits to display, in order. */
+  readonly commits: ReadonlyArray<string>
+
+  /** The commits loaded, keyed by their full SHA. */
+  readonly commitLookup: Map<string, Commit>
+
+  /** The selected commit to style distinctly in the list */
   readonly selectedSHA: string | null
+
+  /** The lookup for GitHub users related to this repository */
   readonly gitHubUsers: Map<string, IGitHubUser>
+
+  /** The emoji lookup to render images inline */
   readonly emoji: Map<string, string>
+
+  /** The list of known local commits for the current branch */
   readonly localCommitSHAs: ReadonlyArray<string>
+
+  /** Callback which fires when a commit has been selected in the list */
+  readonly onCommitChanged: (commit: Commit) => void
+
+  /** Callback that fires when a scroll event has occurred */
+  readonly onScroll: (start: number, end: number) => void
+
+  /** Callback to fire to revert a given commit in the current repository */
+  readonly onRevertCommit: (commit: Commit) => void
+
+  /** Callback to fire to open a given commit on GitHub */
+  readonly onViewCommitOnGitHub: (sha: string) => void
 }
 
 /** A component which displays the list of commits. */
 export class CommitList extends React.Component<ICommitListProps, {}> {
   private renderCommit = (row: number) => {
-    const sha = this.props.history[row]
-    const commit = this.props.commits.get(sha)
+    const sha = this.props.commits[row]
+    const commit = this.props.commitLookup.get(sha)
 
-    if (!commit) {
+    if (commit == null) {
+      if (__DEV__) {
+        log.warn(
+          `[CommitList]: the commit '${sha}' does not exist in the cache`
```

Comment:
While extending this component I realised I'd forgotten to load the commits into the cache from Git (the full commit, not just the SHA).

This should prevent others from making the same mistake.